Shared/Realm.Shared/Linq/RealmResultsVisitor.cs

```diff
@@ -331,14 +333,23 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                 }
             }
 
-            if (m.Method.DeclaringType == typeof(string))
+            if (m.Method.DeclaringType == typeof(string) ||
+                m.Method.DeclaringType == typeof(StringExtensions))
```

Comment:
Initially I had it in an `else if`, but felt a lot of code was duplicated that way so moved it inside with the other string methods.

I'm not sure what's the argument about using it less frequently?